src/ar_storage.erl

```diff
@@ -385,45 +391,22 @@ lookup_tx_filename(ID) ->
 enough_space(Bytes) ->
 	(ar_meta_db:get(disk_space)) >= (Bytes + ar_meta_db:get(used_space)).
 
-%% @doc Sum up the sizes of the files written by the Arweave client.
-calculate_used_space() ->
-	DataDir = ar_meta_db:get(data_dir),
-	{ok, CWD} = file:get_cwd(),
-	Dirs = [
-		filename:join(DataDir, ?BLOCK_DIR),
-		filename:join(DataDir, ?ENCRYPTED_BLOCK_DIR),
-		filename:join(DataDir, ?HASH_LIST_DIR),
-		filename:join(DataDir, ?TX_DIR),
-		filename:join(DataDir, ?TX_INDEX_DIR),
-		filename:join(DataDir, ?WALLET_DIR),
-		filename:join(DataDir, ?WALLET_LIST_DIR),
-		filename:join(CWD, ?LOG_DIR)
-	],
-	lists:foldl(
-		fun(Dir, Acc) ->
-			Acc + calculate_used_space(Dir)
-		end,
-		0,
-		Dirs
-	).
+%% @doc Calculate the available space in bytes on the data directory disk.
+calculate_disk_space() ->
+	{_, Size, _} = get_data_dir_disk_data(),
```

Comment:
Why did we move the pattern match outside of `get_data_dir_disk_data`? I guess this tuple is related to `disksup:get_disk_data` so it would be nice to keep that pattern match as close as possible to that disksup call.

Reply:
I use `get_data_dir_disk_data` in both `calculate_used_space` and `calculate_disk_space`, which depend on different items of this tuple, but neither of them needs the first element, what would you say to returning a tuple of two?

Reply:
In that case, maybe it's fine to keep it as it is. I get you do use the same `disk_data` naming.